src/hooks/usePrevious.ts

```diff
@@ -0,0 +1,9 @@
+import { useEffect, useRef } from "react";
+
+export const usePrevious = <T>(value: T) => {
+  const ref = useRef<T>(value);
+  useEffect(() => {
+    ref.current = value;
+  });
+  return ref.current;
+};
```

Comment:
We can just use `usePrevious` from `react-hooks` now that we have that dependency:

- https://github.com/streamich/react-use/blob/master/docs/usePrevious.md